spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/cql/Schema.scala

```diff
@@ -69,6 +69,13 @@ trait StructDef extends Serializable {
   /** Returns the columns that are not present in the structure. */
   def missingColumns(columnsToCheck: Seq[ColumnRef]): Seq[ColumnRef] =
     for (c <- columnsToCheck if !columnByName.contains(c.columnName)) yield c
+  
+  /** Type of the data described by this struct */
+  type ValueRepr <: AnyRef
+  
+  /** Creates new instance of this struct.
+    * Column values must be given in the same order as columnNames */
+  def newInstance(columnValues: Any*): ValueRepr
```

Comment:
it can't be apply, because it is not a companion object and it doesn't create an instance of StructDef.